### PR TITLE
Update Enphase battery documentation

### PR DESCRIPTION
**Pull request by catsmanac**
## Proposed change

Add more information on battery use in Enphase_Envoy installations.

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
  - Added sensors for grid status, load shedding state, on-grid/off-grid switch, and charging control for Encharge/IQ Batteries.
  - Support for changing battery storage mode and setting reserve battery level.
  - Added sensors for battery storage energy metrics and voltage storage.

- **Enhancements**
  - Expanded capabilities of current transformers with net consumption and production status flags.
  - Improved explanations for system configurations, including Enpower/IQ System Controller and multi-phase installations.
  - Updated authentication requirements and configuration options for Envoy integration.

### CODE REVIEW

File: source/_integrations/enphase_envoy.markdown [hunk 1]
```diff
@@ -48,26 +50,32 @@ For Envoy S Metered / IQ Gateway Metered with installed and configured current t
...
 For Enphase Ensemble systems with the Enpower/IQ System Controller and Encharge/IQ Batteries installed, additional features are available:
 
 - Sensors for battery status and usage
-- Sensors for grid status
-- Sensors for the state of the Enpower's 4 load-shedding relays
-- A switch allowing you to take your system on-grid and off-grid. Note that the Enpower has a slight delay built-in between receiving these commands and actually switching the system on or off grid.
-- A switch allowing you to enable or disable charging the Encharge/IQ Batteries from the power grid.
+- Sensors for grid status (*)
+- Sensors for the state of the Enpower's 4 load-shedding relays (*)
+- A switch allowing you to take your system on-grid and off-grid. Note that the Enpower has a slight delay built-in between receiving these commands and actually switching the system on or off grid. (*)
```
[the remark this answers is not in the document]
Comment (catsmanac) [reply source/_integrations/enphase_envoy.markdown:65]: Enpower is a name.

File: source/_integrations/enphase_envoy.markdown [hunk 2]
```diff
@@ -38,6 +38,8 @@ This integration will offer various sensors depending on the configuration of yo
 
 _Consumption sensors require your Envoy to be properly configured with consumption <abbr title="current transformers">CT</abbr> sensors installed._
 
+### With Current Transformers
```
Comment (c0ffeeca7) [inline source/_integrations/enphase_envoy.markdown]:
```suggestion
### With current transformers
```

File: source/_integrations/enphase_envoy.markdown [hunk 3]
```diff
@@ -48,26 +50,34 @@ For Envoy S Metered / IQ Gateway Metered with installed and configured current t
...
   - Metering status for net consumption and production <abbr title="current transformers">CT</abbr> (`normal` | `not-metering` | `check-wiring`) (aggregate and phase).
-  - Meter status flags active for net consumption and production <abbr title="current transformers">CT</abbr> (aggregate and phase).[^2]
+  - Count of meter status flags active for net consumption and production <abbr title="current transformers">CT</abbr> (aggregate and phase).[^2]
 
-[^1]: For multi-phase systems, the Envoy sums the voltages of the phases. May be valid for split-phase, but for 3-phase systems use the individual phases rather than the summed value.
+[^1]: For multiphase systems, the Envoy sums the voltages of the phases. This may be valid for split-phase, but for 3-phase systems, use the individual phases rather than the summed value.
 [^2]: If this value is non-zero, consult the diagnostic report of the Envoy and look for `raw_data` - `/ivp/meters` - `statusFlags` for set flags (`production-imbalance` | `negative-production` | `power-on-unused-phase` | `negative-total-consumption`).
 
+### With Batteries
```
Comment (c0ffeeca7) [inline source/_integrations/enphase_envoy.markdown]:
```suggestion
### With batteries
```

File: source/_integrations/enphase_envoy.markdown [hunk 4]
```diff
@@ -48,26 +50,34 @@ For Envoy S Metered / IQ Gateway Metered with installed and configured current t
...
 
 - Sensors for battery status and usage
-- Sensors for grid status
-- Sensors for the state of the Enpower's 4 load-shedding relays
-- A switch allowing you to take your system on-grid and off-grid. Note that the Enpower has a slight delay built-in between receiving these commands and actually switching the system on or off grid.
-- A switch allowing you to enable or disable charging the Encharge/IQ Batteries from the power grid.
+- Sensors for grid status (*)
+- Sensors for the state of the Enpower's four load-shedding relays (*)
+- A switch allowing you to take your system on-grid and off-grid. Note that the Enpower has a slight delay built-in between receiving these commands and actually switching the system on or off grid. (*)
+- A switch allowing you to enable or disable charging the Encharge/IQ Batteries from the power grid. (*)
```
Comment (c0ffeeca7) [inline source/_integrations/enphase_envoy.markdown]:
```suggestion
- A switch allowing you to enable or disable charging the Encharge/IQ batteries from the power grid. (*)
```

File: source/_integrations/enphase_envoy.markdown [hunk 5]
```diff
@@ -48,26 +50,34 @@ For Envoy S Metered / IQ Gateway Metered with installed and configured current t
...
+
+In multiphase installations with batteries, in countries with phase-balancing grid meters, the battery will export to the grid on one phase the amount it lacks on another phase, as if it is using the grid as a 'transport' between phases. Since the grid meter will balance the amount in and exported on the two phases, the net result is zero. The Envoy multiphase consumption CTs, however, will report the amounts on both phases, resulting in too high export on one and too high import on the other. One needs to use the sum of grid import and export to eliminate this effect.
+
+In multiphase installations with batteries, in countries with phase-balancing grid meters, the battery will export to the grid on one phase the amount it lacks on another phase, as if it is using the grid as a 'transport' between phases. Since the grid meter will balance the amount in and exported on the two phases, the net result is zero. The Envoy multiphase consumption CTs, however, will report the amounts on both phases, resulting in too high export on one and too high import on the other. One needs to use the sum of grid import and export to eliminate this effect.
```
Comment (c0ffeeca7) [inline source/_integrations/enphase_envoy.markdown]:
```suggestion

```
I think this is a duplicate

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @catsmanac :+1:

**Pull request merged**